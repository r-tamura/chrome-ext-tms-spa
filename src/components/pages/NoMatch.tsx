import * as React from "react";

export const NoMatch: React.FC = () => {
  return (
    <main className="main">
      <h1>Not Found 404</h1>
    </main>
  );
};
