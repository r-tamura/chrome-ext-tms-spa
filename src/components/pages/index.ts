export * from "./Attendance";
export * from "./SignInPage";
export * from "./TransportExpensePage";
export * from "./NoMatch";
export * from "./App";
export * from "./Root";
