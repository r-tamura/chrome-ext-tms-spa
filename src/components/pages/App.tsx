import React from "react";
import { Helmet } from "react-helmet";
import { Route, Switch, Redirect } from "react-router-dom";
import { AppHeader, Nav } from "~/components/organisms";
import { EnsureLoggedInContainer } from "~/components/containers/EnsureLoggedInContainer";
import { TransportExpensePage, TimeCardPage, SignInPage, NoMatch } from ".";
import { AppGrid } from "~/components/organisms";

const PAGES_WITH_NAVBAR = ["/transportation", "/attendance"];

export function App() {
  return (
    <AppGrid>
      <Helmet>
        <link
          href="https://fonts.googleapis.com/css?family=M+PLUS+Rounded+1c"
          rel="stylesheet"
        ></link>
        <meta
          name="viewport"
          content="width=device-width, initial-scale=1.0, shrink-to-fit=no"
        />
      </Helmet>
      <AppHeader />
      <Route
        path={PAGES_WITH_NAVBAR}
        render={renderProps => {
          return <Nav path={renderProps.match.url} />;
        }}
      />
      <Switch>
        <Route
          exact={true}
          path="/"
          component={() => <Redirect to={"/signin"} />}
        />
        <Route exact={true} path="/signin" component={SignInPage} />
        <EnsureLoggedInContainer>
          {/* TODO: Dashboard画面作成までは交通費管理画面へリダイレクト */}
          <Route
            path="/dashboard"
            component={() => <Redirect to={"/transportation"} />}
          />
          <Route path="/transportation" component={TransportExpensePage} />
          <Route path="/attendance" component={TimeCardPage} />
        </EnsureLoggedInContainer>
        <Route component={NoMatch} />
      </Switch>
    </AppGrid>
  );
}
