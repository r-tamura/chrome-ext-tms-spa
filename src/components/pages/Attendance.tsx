import React from "react";
import styled from "styled-components";
import { Helmet } from "react-helmet";
import { SelectBox } from "~/components/SelectBox";
import { Button, ButtonGroup } from "~/components/atoms";
import { zerofill } from "~/helpers/common";
import { AttendanceMonthlyView } from "~/types";
import { MsgAttendance } from "~/helpers/_const";
import { useAttendance } from "~/stores/hooks";
import { MainContent } from "~/components/organisms";
import { BOLD } from "~/styles/font";

export const AttendancePage: React.FC = () => {
  const {
    projects,
    fetchAttendances,
    attendanceSettings,
    updateSettings,
    attendanceMonthly,
    saveAttendances,
    updateDaily,
    changeMonth,
    setMonthlyWithDefaults
  } = useAttendance();

  const { year, month } = attendanceMonthly;
  return (
    <main className="app-content-with-navbar">
      <Helmet>
        <title>Attendance | TMS</title>
      </Helmet>
      <MainContent>
        <h1>
          <Button id={"btn-prev-month"} onClick={onPrevMonthClick}>
            {"<"}
          </Button>
          {year} / {month}
          <Button id={"btn-next-month"} onClick={onNextMonthClick}>
            {">"}
          </Button>
        </h1>
        {renderMonthly(attendanceMonthly)}

        {/* 時間リスト */}
        <datalist id="timelist-am">
          <option>09:00</option>
          <option>09:30</option>
          <option>10:00</option>
          <option>10:30</option>
        </datalist>
        <datalist id="timelist-pm">
          <option>17:30</option>
          <option>18:00</option>
          <option>18:30</option>
          <option>19:00</option>
          <option>19:30</option>
          <option>19:30</option>
          <option>20:00</option>
          <option>20:30</option>
          <option>21:00</option>
          <option>21:30</option>
          <option>22:00</option>
          <option>22:30</option>
          <option>23:00</option>
          <option>23:30</option>
        </datalist>
      </MainContent>
    </main>
  );

  function renderMonthly({
    days,
    isFetching,
    hasApplied
  }: AttendanceMonthlyView) {
    if (isFetching) {
      // TODO: ロードインジケータを表示するように
      return <p>Now Loading...</p>;
    }

    if (days.length === 0) {
      return <p>勤怠データを取得できません</p>;
    }

    return (
      <div>
        <ButtonGroup>
          <Button id={"btn-set-default"} onClick={onSetDefaultClick}>
            set default
          </Button>
          <Button id={"btn-save"} onClick={onSaveClick}>
            Save
          </Button>
          <Button id={"btn-fetch"} onClick={fetchAttendances}>
            Reload
          </Button>
          {hasApplied ? (
            <Button id={"btn-submit"} title={"上長申請中"} disabled>
              Submit
            </Button>
          ) : (
            <Button id={"btn-submit"} title={"上長申請"} onClick={onSubmit}>
              Submit
            </Button>
          )}
        </ButtonGroup>
        <div>
          {/* <SelectBox
            value={attendanceSettings.projectId || ""}
            options={{
              items: projects,
              valueKey: "projectId",
              labelKey: "name"
            }}
            onChange={$select =>
              updateSettings({
                projectId: $select.selectedOptions[0].value
              })
            }
          /> */}
          select box
        </div>
        <table className="tms-table tms-table--bordered">
          <thead>
            <tr>
              <th>日</th>
              <th>出勤</th>
              <th>退勤</th>
              {/* <th>合計</th>
              <th>Extra</th>
              <th>MN Extra</th> */}
              <th>プロジェクト</th>
              <th>承認</th>
            </tr>
          </thead>
          <tbody>
            {days.map((a, index) => (
              <tr
                key={a.dailyId}
                className={a.hasUpdated ? "attendance-diff" : ""}
              >
                {/* <td
                  className={
                    a.isWeekday ? "attendance-weekday" : "attendance-holiday"
                  }
                >
                  {zerofill(2, index + 1)}
                </td> */}
                {a.isWeekday ? (
                  <TdWeekday>{zerofill(2, index + 1)}</TdWeekday>
                ) : (
                  <TdHoliday>{zerofill(2, index + 1)}</TdHoliday>
                )}
                <td className="tms-textfield tms-textfield--table">
                  <input
                    type="time"
                    value={a.start}
                    list="timelist-am"
                    onChange={e =>
                      updateDaily(a.dailyId, {
                        start: e.target.value
                      })
                    }
                  />
                </td>
                <td className="tms-textfield tms-textfield--table">
                  <input
                    type="time"
                    value={a.end}
                    list="timelist-pm"
                    onChange={e =>
                      updateDaily(a.dailyId, { end: e.target.value })
                    }
                  />
                </td>
                {/* プロジェクトが未設定の場合は表示なし */}
                <td>
                  {/* <SelectBox
                    value={a.project ? a.project.projectId : ""}
                    options={{
                      items: projects,
                      valueKey: "projectId",
                      labelKey: "name"
                    }}
                    onChange={$select =>
                      updateDaily(a.dailyId, {
                        projectId: $select.selectedOptions[0].value
                      })
                    }
                  // additionalClass={"tms-select--table"}
                  /> */}
                  select box
                </td>
                <td>
                  {a.hasConfirmed
                    ? MsgAttendance.HAS_SUBMITED
                    : MsgAttendance.NOT_SUBMITED}
                </td>
              </tr>
            ))}
          </tbody>
        </table>
      </div>
    );
  }

  function onSaveClick(): void {
    saveAttendances();
  }

  function onSetDefaultClick(): void {
    setMonthlyWithDefaults(attendanceMonthly.days.map(d => d.dailyId));
  }

  function onNextMonthClick(): void {
    const { year: curYear, month: curMonth } = attendanceMonthly;
    const nextMonth = (curMonth % 12) + 1;
    const nextYear = nextMonth === 1 ? curYear + 1 : curYear;
    changeMonth(nextYear, nextMonth);
  }

  function onPrevMonthClick(): void {
    const { year: curYear, month: curMonth } = attendanceMonthly;
    const prevMonth = ((curMonth + 10) % 12) + 1;
    const prevYear = prevMonth === 12 ? curYear - 1 : curYear;
    changeMonth(prevYear, prevMonth);
  }

  function onSubmit(): void {
    console.log("[Attendance] Submitted");
  }
};

const TdHoliday = styled.td`
  color: rgba(255, 0, 0, 0.4);
  font-weight: ${BOLD};
`;

const TdWeekday = styled.td``;
