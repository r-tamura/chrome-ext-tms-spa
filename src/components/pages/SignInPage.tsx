import React, { useEffect } from "react";
import { Helmet } from "react-helmet";
import { Form, Text } from "react-form";
import styled from "styled-components";
import { Panel, Button } from "~/components/atoms";
import { MQ_S } from "~/styles/mediaquery";
import { useUser, useRouter } from "~/stores/hooks";
import { FormItemContainer } from "~/components/atoms";

/**
 * ログインページ
 */
export const SignInPage: React.FC = () => {
  const { isAuthenticated, isFetching, login } = useUser();
  const { goto } = useRouter();

  useEffect(() => {
    function redirectToDashboardIfNeeded(isAuthenticated: boolean) {
      if (shouldRedirectToDashBoard(isAuthenticated)) {
        goto("transportation");
      }
    }
    redirectToDashboardIfNeeded(isAuthenticated);
  }, [goto, isAuthenticated]);

  if (isFetching) {
    return <p>Fetching...</p>;
  }

  return (
    <Root className="app-content-full">
      <Helmet>
        <title>Sign In | TMS</title>
      </Helmet>
      <LoginMain>
        <h1>Sign in to TMS</h1>
        <p>
          Enter your <strong>user name</strong> and <strong>password</strong>.
        </p>
        <Form onSubmit={handleSubmit}>
          {formApi => (
            <Panel>
              <form onSubmit={formApi.submitForm}>
                {/* User name text filed */}
                <FormItemContainer>
                  <Text
                    field={"username"}
                    id={"username"}
                    placeholder="John Doe"
                    maxLength={30}
                    required={true}
                  />
                  <label htmlFor={"username"}>UserName</label>
                </FormItemContainer>
                {/* Password text field */}
                <FormItemContainer>
                  <Text
                    type="password"
                    field={"password"}
                    id={"password"}
                    placeholder="Your password"
                    maxLength={30}
                    required={true}
                  />
                  <label htmlFor={"password"}>Password</label>
                </FormItemContainer>
                <Button variant="contained" color="primary" block>
                  Sign in
                </Button>
              </form>
            </Panel>
          )}
        </Form>
      </LoginMain>
    </Root>
  );

  function shouldRedirectToDashBoard(isAuthenticated: boolean): boolean {
    return isAuthenticated;
  }

  function handleSubmit({ username = "", password = "" }) {
    login(username, password);
  }
};

const Root = styled.main`
  align-self: center;
  justify-self: center;
`;

const LoginMain = styled.div`
  width: 95%;
  align-self: center;
  justify-self: center;

  & > * {
    text-align: center;
  }

  ${MQ_S} {
    width: 550px;
  }
`;
