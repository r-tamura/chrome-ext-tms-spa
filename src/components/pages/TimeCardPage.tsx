import React from "react";
import styled from "styled-components";
import { Helmet } from "react-helmet";
import { SelectBox } from "~/components/SelectBox";
import { Button, ButtonGroup, Input } from "~/components/atoms";
import { zerofill } from "~/helpers/common";
import { AttendanceMonthlyView, Project } from "~/types";
import { MsgAttendance } from "~/helpers/_const";
import { useAttendance } from "~/stores/hooks";
import { MainContent } from "~/components/organisms";
import { BOLD } from "~/styles/font";
import { Table, Tr, Td } from "../atoms/Table";

export const TimeCardPage: React.FC = () => {
  const {
    projects,
    fetchAttendances,
    attendanceSettings,
    updateSettings,
    attendanceMonthly,
    saveAttendances,
    updateDaily,
    setMonthlyWithDefaults,
    submitApplication,
    nextMonth,
    prevMonth
  } = useAttendance();

  const { year, month } = attendanceMonthly;
  return (
    <main className="app-content-with-navbar">
      <Helmet>
        <title>Attendance | TMS</title>
      </Helmet>
      <MainContent>
        <h1>
          <Button id={"btn-prev-month"} onClick={handlePrevMonthClick}>
            {"<"}
          </Button>
          <span>
            {year} / {month}
          </span>
          <Button id={"btn-next-month"} onClick={handleNextMonthClick}>
            {">"}
          </Button>
        </h1>
        {renderMonthly(attendanceMonthly)}

        {/* 時間リスト */}
        <datalist id="timelist-am">
          <option>09:00</option>
          <option>09:30</option>
          <option>10:00</option>
          <option>10:30</option>
        </datalist>
        <datalist id="timelist-pm">
          <option>17:30</option>
          <option>18:00</option>
          <option>18:30</option>
          <option>19:00</option>
          <option>19:30</option>
          <option>19:30</option>
          <option>20:00</option>
          <option>20:30</option>
          <option>21:00</option>
          <option>21:30</option>
          <option>22:00</option>
          <option>22:30</option>
          <option>23:00</option>
          <option>23:30</option>
        </datalist>
      </MainContent>
    </main>
  );

  function renderMonthly({
    days,
    isFetching,
    hasApplied
  }: AttendanceMonthlyView) {
    if (isFetching) {
      // TODO: ロードインジケータを表示するように
      return <p>Now Loading...</p>;
    }

    if (days.length === 0) {
      return <p>勤怠データを取得できません</p>;
    }

    return (
      <div>
        <ButtonGroup>
          <Button
            id={"btn-set-default"}
            variant={"contained"}
            color={"default"}
            onClick={handleSetDefaultClick}
          >
            set default
          </Button>
          <Button
            id={"btn-force-fetch"}
            variant={"contained"}
            color={"default"}
            onClick={fetchAttendances}
          >
            Reload
          </Button>
          <Button
            id={"btn-save"}
            variant={"contained"}
            color={"primary"}
            onClick={handleSaveClick}
          >
            Save
          </Button>
          {hasApplied ? (
            <Button
              id={"btn-submit"}
              variant={"contained"}
              color={"secondary"}
              title={"上長申請中"}
              disabled
            >
              Submit
            </Button>
          ) : (
            <Button
              id={"btn-submit"}
              variant={"contained"}
              color={"secondary"}
              title={"上長申請"}
              onClick={onSubmit}
            >
              Submit
            </Button>
          )}
        </ButtonGroup>
        <div>
          <SelectBox
            name="projectId"
            label="デフォルトプロジェクト"
            defaultValue={
              attendanceSettings.projectId ? attendanceSettings.projectId : ""
            }
            options={{
              items: projects,
              value: (project: Project) => project.projectId,
              label: (project: Project) => project.name
            }}
            onChange={e => {
              updateSettings({
                projectId: e.currentTarget.selectedOptions.item(0).value
              });
            }}
          />
        </div>
        <Table header={["日", "出勤", "退勤", "プロジェクト", "承認"]}>
          {days.map((record, index) => (
            <Tr
              key={record.dailyId}
              className={record.hasUpdated ? "attendance-diff" : ""}
            >
              {record.isWeekday ? (
                <TdWeekday>{zerofill(2, index + 1)}</TdWeekday>
              ) : (
                <TdHoliday>{zerofill(2, index + 1)}</TdHoliday>
              )}
              <Td>
                <Input
                  type="time"
                  value={record.start}
                  list="timelist-am"
                  onChange={e =>
                    updateDaily(record.dailyId, {
                      start: e.currentTarget.value
                    })
                  }
                />
              </Td>
              <Td>
                <Input
                  type="time"
                  value={record.end}
                  list="timelist-pm"
                  onChange={e =>
                    updateDaily(record.dailyId, { end: e.currentTarget.value })
                  }
                />
              </Td>
              {/* プロジェクトが未設定の場合は表示なし */}
              <Td>
                <SelectBox
                  name="projectId"
                  defaultValue={record.project ? record.project.projectId : ""}
                  options={{
                    items: projects,
                    value: (project: Project) => project.projectId,
                    label: (project: Project) => project.name
                  }}
                  onChange={e =>
                    updateDaily(record.dailyId, {
                      projectId: e.currentTarget.selectedOptions.item(0).value
                    })
                  }
                  // ref={register({ required: true })}
                />
              </Td>
              <Td>
                {record.hasConfirmed
                  ? MsgAttendance.HAS_SUBMITED
                  : MsgAttendance.NOT_SUBMITED}
              </Td>
            </Tr>
          ))}
        </Table>
      </div>
    );
  }

  function handleSaveClick() {
    saveAttendances();
  }

  function handleSetDefaultClick() {
    setMonthlyWithDefaults(attendanceMonthly.days.map(d => d.dailyId));
  }

  function handleNextMonthClick() {
    nextMonth();
  }

  function handlePrevMonthClick() {
    prevMonth();
  }

  function onSubmit() {
    console.log(`[Attendance] Submitted ${year}/${month}`);
    // const { year, month } = attendanceMonthly;
    // submitApplication(year, month);
  }
};

const TdHoliday = styled.td`
  color: rgba(255, 0, 0, 0.4);
  font-weight: ${BOLD};
`;

const TdWeekday = styled.td``;
