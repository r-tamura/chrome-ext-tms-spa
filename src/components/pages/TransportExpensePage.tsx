import React from "react";
import Helmet from "react-helmet";
import {
  TransExpenseHistory,
  TransExpenseTemplates
} from "~/components/organisms";
import { useTranspotation } from "~/stores/hooks";
import { MainContent } from "~/components/organisms";

export const TransportExpensePage: React.FC = () => {
  const {
    /* states */
    expenses,
    projects,
    usages,
    objectives,
    templates,
    /* dispatchers */
    createExpenseTemplate,
    deleteExpenseTemplate,
    updateExpenseTemplate,
    createExpense,
    createExpenseFromTemplate,
    updateExpense,
    deleteExpense
  } = useTranspotation();

  return (
    <main className="app-content-with-navbar">
      <Helmet>
        <title>Transportation | TMS</title>
      </Helmet>
      <MainContent>
        <h1>交通費テンプレート</h1>
        <TransExpenseTemplates
          templates={templates}
          projects={projects}
          usages={usages}
          objectives={objectives}
          createExpenseTemplate={createExpenseTemplate}
          deleteExpenseTemplate={deleteExpenseTemplate}
          updateExpenseTemplate={updateExpenseTemplate}
        />
        <h1>交通費履歴</h1>
        <TransExpenseHistory
          expenses={expenses}
          templates={templates}
          projects={projects}
          usages={usages}
          objectives={objectives}
          createExpense={createExpense}
          deleteExpense={deleteExpense}
          updateExpense={updateExpense}
          createExpenseFromTemplate={createExpenseFromTemplate}
        />
      </MainContent>
    </main>
  );
};
