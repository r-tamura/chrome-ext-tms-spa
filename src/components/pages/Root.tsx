import React from "react";
import { Store, AnyAction } from "redux";
import { Provider as ReduxProvider } from "react-redux";
import { ConnectedRouter } from "connected-react-router";
import { ThemeProvider } from "styled-components";
import { history } from "~/stores";
import { App } from "~/components/pages";
import { RootState } from "~/modules";
import { telema } from "~/styles/theme";
import { GlobalStyle } from "~/styles/GlobalStyle";

interface Props {
  store: Store<RootState, AnyAction>;
}

export const Root: React.FC<Props> = ({ store }) => {
  return (
    <ReduxProvider store={store}>
      <ThemeProvider theme={telema}>
        <ConnectedRouter history={history}>
          <>
            <App />
            <GlobalStyle />
          </>
        </ConnectedRouter>
      </ThemeProvider>
    </ReduxProvider>
  );
};
