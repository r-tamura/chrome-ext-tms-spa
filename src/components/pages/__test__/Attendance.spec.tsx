import * as React from "react";
import renderer from "react-test-renderer";
import { shallow, mount } from "enzyme";
import { AttendancePage } from "~/components/pages";
import {
  AttendanceSettingsModelBuilder,
  AttendanceViewModelBuilder,
  ProjectsBuilder
} from "../../../../test/__mocks__";
import * as hooks from "~/stores/hooks";
import { AppThemeProvider } from "~/components/Provider";

jest.mock("../../../stores/hooks");
const mockedHooks = hooks as jest.Mocked<typeof hooks>;

const createComponent = ({
  attendanceMonthly = AttendanceViewModelBuilder.build(),
  attendanceSettings = AttendanceSettingsModelBuilder.build(),
  projects = ProjectsBuilder.of()
    .with2Projects()
    .build(),
  fetchAttendancesIfNeeded = jest.fn(),
  fetchSettings = jest.fn(),
  saveAttendancesIfNeeded = jest.fn(),
  updateDaily = jest.fn(),
  setMonthlyWithDefaults = jest.fn(),
  updateSettings = jest.fn(),
  changeMonth = jest.fn(),
  submitApplication = jest.fn()
} = {}) => {
  mockedHooks.useAttendance.mockReturnValue({
    attendanceMonthly,
    attendanceSettings,
    projects,
    fetchAttendances: fetchAttendancesIfNeeded,
    fetchSettings,
    saveAttendances: saveAttendancesIfNeeded,
    updateDaily,
    setMonthlyWithDefaults,
    updateSettings,
    changeMonth,
    submitApplication
  });
  return {
    fetchAttendancesIfNeeded,
    fetchSettings,
    saveAttendancesIfNeeded,
    updateDaily,
    setMonthlyWithDefaults,
    updateSettings,
    changeMonth,
    submitApplication,
    Component: (
      <AppThemeProvider>
        <AttendancePage />
      </AppThemeProvider>
    )
  };
};

describe("<AttendancePage />", () => {
  it("shoud match existing snapshot", () => {
    const { Component } = createComponent();
    const page = renderer.create(Component).toJSON();
    expect(page).toMatchSnapshot();
  });

  it("should be right next month", () => {
    const { Component } = createComponent();
    const w = shallow(Component);
  });

  it("should make disable submit button", () => {
    const { changeMonth, Component } = createComponent({
      attendanceMonthly: AttendanceViewModelBuilder.of()
        .hasApplied()
        .build()
    });
    const w = mount(Component);

    w.find("button#btn-next-month").simulate("click");
    expect(changeMonth.mock.calls[0]).toEqual([2018, 2]);

    w.find("button#btn-prev-month").simulate("click");
    expect(changeMonth.mock.calls[1]).toEqual([2017, 12]);
    const $submit = w.find("button#btn-submit");
    expect($submit.length).toBe(1);
  });

  // it("should submit Application", () => {
  //   const mock = createComponent({ attendanceMonthly: AttendanceViewModelBuilder.of().hasApplied().build() })
  //   const w = mount(mock.component)
  //   w.find("").simulate("click")
  //   expect()
  // })
});
