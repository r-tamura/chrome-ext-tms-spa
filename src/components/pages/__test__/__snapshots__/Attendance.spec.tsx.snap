// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AttendancePage /> shoud match existing snapshot 1`] = `
<main
  className="app-content-with-navbar"
>
  <div
    className="sc-VigVT iclQvs"
  >
    <h1>
      <button
        className="sc-bdVaJa sc-bxivhb igplhg"
        color="default"
        id="btn-prev-month"
        onClick={[Function]}
      >
        &lt;
      </button>
      2018
       / 
      1
      <button
        className="sc-bdVaJa sc-bxivhb igplhg"
        color="default"
        id="btn-next-month"
        onClick={[Function]}
      >
        &gt;
      </button>
    </h1>
    <div>
      <div
        className="sc-EHOje gwoyUY"
      >
        <button
          className="sc-bdVaJa sc-bxivhb igplhg"
          color="default"
          id="btn-set-default"
          onClick={[Function]}
        >
          set default
        </button>
        <button
          className="sc-bdVaJa sc-bxivhb igplhg"
          color="default"
          id="btn-save"
          onClick={[Function]}
        >
          Save
        </button>
        <button
          className="sc-bdVaJa sc-bxivhb igplhg"
          color="default"
          id="btn-fetch"
          onClick={[MockFunction]}
        >
          Reload
        </button>
        <button
          className="sc-bdVaJa sc-bxivhb igplhg"
          color="default"
          id="btn-submit"
          onClick={[Function]}
          title="上長申請"
        >
          Submit
        </button>
      </div>
      <div>
        select box
      </div>
      <table
        className="tms-table tms-table--bordered"
      >
        <thead>
          <tr>
            <th>
              日
            </th>
            <th>
              出勤
            </th>
            <th>
              退勤
            </th>
            <th>
              プロジェクト
            </th>
            <th>
              承認
            </th>
          </tr>
        </thead>
        <tbody>
          <tr
            className=""
          >
            <td
              className="sc-kpOJdX kEtjJz"
            >
              01
            </td>
            <td
              className="tms-textfield tms-textfield--table"
            >
              <input
                list="timelist-am"
                onChange={[Function]}
                type="time"
                value="09:00"
              />
            </td>
            <td
              className="tms-textfield tms-textfield--table"
            >
              <input
                list="timelist-pm"
                onChange={[Function]}
                type="time"
                value="17:30"
              />
            </td>
            <td>
              select box
            </td>
            <td>
              承認
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <datalist
      id="timelist-am"
    >
      <option>
        09:00
      </option>
      <option>
        09:30
      </option>
      <option>
        10:00
      </option>
      <option>
        10:30
      </option>
    </datalist>
    <datalist
      id="timelist-pm"
    >
      <option>
        17:30
      </option>
      <option>
        18:00
      </option>
      <option>
        18:30
      </option>
      <option>
        19:00
      </option>
      <option>
        19:30
      </option>
      <option>
        19:30
      </option>
      <option>
        20:00
      </option>
      <option>
        20:30
      </option>
      <option>
        21:00
      </option>
      <option>
        21:30
      </option>
      <option>
        22:00
      </option>
      <option>
        22:30
      </option>
      <option>
        23:00
      </option>
      <option>
        23:30
      </option>
    </datalist>
  </div>
</main>
`;
